tasks.py

```diff
@@ -0,0 +1,35 @@
+# Copyright Contributors to the OpenImageIO project.
+# SPDX-License-Identifier: Apache-2.0
+# https://github.com/AcademySoftwareFoundation/OpenImageIO
+
+from pathlib import Path
+import shutil
+import invoke
+from repairwheel._vendor.auditwheel.wheeltools import InWheelCtx
+
+
+DIRS_TO_REMOVE = ['lib', 'lib64', 'share', 'include']
+
+def _wheel_edit(wheel_path, dirs_to_remove=DIRS_TO_REMOVE):
```

Comment:
I think we can avoid this by using cmake components. In OpenEXR, we did https://github.com/AcademySoftwareFoundation/openexr/commit/222bc3d44959b8eabd135f6cb4c0adffee835394. Basically, the cmake build system should only install what's needed. That's the most reliable way to manage this and also the recommended way from the scikit-build-core team.

Reply:
Yeah, I noticed you'd done this with OpenEXR...

The real reason for doing what I'm doing here is to get rid of the versioned copies of the libraries -- libOpenImageIO.so.2.6, etc. -- cuz the symlinks get copied over to the wheel as discrete files. It just happened to be a convenient place to strip out everything else at the same time, too.

Would using components prevent those versioned copies from getting created?

The other thing is, we're building libOpenImageIO as a dynamic library, so it can be shared by the Python module and the CLI tools; and we're also building OpenColorIO as a dynamic library for the time being "to prevent problems downstream" (Larry can elaborate). I think in the end, we'd only use components to shave off the headers, CMake configs, and fonts -- and even those things are things I'd like to provide some day as part of the bdist, if we can correct the CMake configs and provide helpers for downstream builds.

If we can get rid of the "versioned copies" of the libraries, whether by using components or some other method, we could skip the repair-wheel step entirely.

Reply:
>Would using components prevent those versioned copies from getting created?

No, for that you'd have to if/else setting SOVERSION. [Here](https://github.com/AcademySoftwareFoundation/OpenImageIO/blob/a946288db47df47b257bf79732a0de40ccf1adae/src/libutil/CMakeLists.txt#L99-L100) and [here](https://github.com/AcademySoftwareFoundation/OpenImageIO/blob/a946288db47df47b257bf79732a0de40ccf1adae/src/libOpenImageIO/CMakeLists.txt#L218-L219) I think.

Using components would only allow you to only install what you want and nothing else. That with not setting VERSION and SOVERSION I think is the cleanest and most future proof and easy to maintain option.